### DISCUSSION TITLE
multiple turbines simulation use OLAF

### DISCUSSION THREAD

**Discussion by robytian** [OpenFAST/openfast#1799] [Q&A] [answered]
Hi @ebranlard 

I compiled the Aerodyn_driver code and ran the ad_MultipleHAWT test with OLAF method. There are some questions I would like to ask you：

1.I have found by simulating multiple turbines that the vortex filaments produced by different turbines that are closer to each other will deform. I'm guessing that you must have considered the vortex systems produced by different turbines inducing each other. Is that correct?

2.If mutual induction between different turbines is taken into account, then the vortex filaments produced by different turbines will also have an effect on the bound circulation on the blades of other turbines, which is also taken into account right?

3.I found that I only needed to enter more geometry, motion, and other information about the turbines in the input file to complete the construction of a multi-unit wind farm. I would like to ask how this is achieved that the vortex systems generated by different turbines can be induced to each other across the field?

Thanks a lot.

Roby

**Comment by ebranlard** [comment, marked as answer]
Hi, 

The short answer to all your questions, is that indeed OLAF considers all the vorticity at once, should it come from one or multiple rotors. All the vortex filaments (from all the rotors) influence each other, and each blade will feel the influences of all the vortex filaments (including other blades and other turbines wakes).  

You could simulate wind farms as well with the AeroDyn driver, it willl be slightly expensive (due to the long wake required), but it's possible. It is important to note that our vortex filament method does not account for viscosity fully, and therefore, far wakes might not be accurate. If you have turbulence, this lack of diffusion is not as problematic as the meandering of the wake will be the primary effect.  (see the[ OLAF documentation guidelines](https://openfast.readthedocs.io/en/dev/source/user/aerodyn-olaf/RunningOLAF.html)for more details) 

So th answers are: 
1. Yes
2. Yes
3. Yes you can simulate wind farms. I would recommend reading the [theory section of OLAF](https://openfast.readthedocs.io/en/dev/source/user/aerodyn-olaf/OLAFTheory.html) or this [reference ](https://link.springer.com/book/10.1007/978-3-319-55164-7) for more details. OLAF distinguishes between free vorticity and bound vorticity, but whether it comes from one or multiple rotors does not affect the theory. 

I hope that helps.

**Comment by robytian** [comment]
Hi ebranlard

I'm trying to understand exactly what is meant for the division of the wake panels, do you see if I'm understanding correctly?

1.The NWFree consider real-time induced velocities for all shedding and trailing vortex, including vortex core growth models. Perhaps there are vortex  models such as vortex stretching?

2.NWFrozen is after a certain amount of NWFree panels is reached, this velocity becomes the then-last-minute velocity for convection, and the vortex core growth model is no longer considered? The velocity due to the NWFrozen panels does not change after NWFree?

3.Is the FWFree to continue to consider the vortex core growth model after the end of the NWFrozen at the same velocity as the NWFree and solve for the induced velocity? At this point, only the tip vortex filament and the root vortex filament (trailing vortex) are considered, and is that where these two vortex filaments are generated at the tip and root of the blade? Are their vortices redistributed or are the original NWFree circulation corresponding to the tip and root used directly?

4.FWFrozen is the continuation of convection at an average velocity after a certain number of FWFree panels have been reached, how are the velocities behind this affected?

I'm not sure if I'm understanding this correctly, perhaps you could inform me how it is taken into account in the recalculation of the induced velocities of the different wake panels once they reach a certain number?

Thank you so much！

Roby

**Reply by ebranlard** [reply]
Here are some responses to your questions:

1a. That is correct. 
1b. Vortex core stretching is not yet implemented. 

2a. The velocity used in the "frozen" part of the wake is not the "last-minute" convection velocity, but an average over over a zone of the Near wake. It is computed in this zone at every time step. You can see this part of the [documentation ](https://openfast.readthedocs.io/en/dev/source/user/aerodyn-olaf/OLAFTheory.html#frozen-vorticity-convection)for more details.  The frozen velocity change at each time step, but it will be the same for all the elements in the frozen wake. Ideally, we would keep track of the history of the frozen velocity, but we currently don't do that and move the entire frozen wake with one velocity only.  
2b. The core growth model is still applied in the frozen wake. 

3a. You cannot have a FWFree if you have a NWFrozen (OLAF will abort). You can have a FWFrozen though. The core growth will also be applied in the FWFrozen. 
3b. The tip and root vorticity positions are determined such as it conserves the first moment of vorticity (  r * Gamma(r)  )  in the upper half and lower half of the blade wake. What this means is that, if for some reason your strongest trailed vorticity is at r=0.3R and not r=0, then OLAF will shed a far wake vortex at r=0.3R (instead of r=0). The same is true for the tip vorticity, it will not necessarily be at r=R (but typically it will be very close to it).  
3c. The intensity of the far wake vortices corresponds to the total circulation of the last NW panel. In the discretized world, this is simply the sum of the Gamma's of the panels. 

4. Hopefully the documentation mentioned above will help clarify. The behavior for the FWFrozen and NWFrozen is the same.

**Comment by robytian** [comment]
Hi @ebranlard

In the Aerodyn_driver module, the ad_MultipleHAWT test, if wakemethod selects 1=BEMT or 2=DBEMT, how is the induced velocity of the downstream turbine taken into account? How is the downstream turbine affected by the wake flow of the upstream turbine? As far as I understand, the BEM cannot solve for the velocity field for the time being but rather updates the blade induced velocities through the induction factor.

Thanks!

Roby

**Reply by jjonkman** [reply]
Dear @robytian,

If you choose to use BEM or DBEMT with the AeroDyn driver supporting multiple wind turbines, the aerodynamics of each wind turbine are treated independently (without coupling through wake effects).  The use of OLAF is preferred if the multiple wind turbines are in close proximity.

Best regards,

**Comment by robytian** [comment]
Hi @ebranlard @jjonkman 

I modeled a 5MW turbine using the latest version of Aerodyn_driver and did the calculations using BEM and OLAF respectively and found the following problems:
1.Why does the strength of the circulation in different blade segments of the blade differ for the two methods?
2.Why is the circulation distribution of the blade periodic, rather than regionally a stable value? Because my simulation is constant. for OLAF panel respectively I gave 500, 500, 1000, 1000 which should be enough.
3.The circulation on the blade is always conserved, for example, if I need to get the distribution of the circulation, is it ok to take the value at any moment after stabilization or do I need to do an average?
4.The aerodynamic power, thrust and torque of the BEM and OLAF are calculated differently and are somewhat different, what causes this? And can such calculations be used?
Thanks all ！

![1](https://github.com/OpenFAST/openfast/assets/86820784/932ac3e8-eeaf-4124-b62d-7b6d04aca10c)
OLAF circulation distribution
![2](https://github.com/OpenFAST/openfast/assets/86820784/3f6886bb-1177-4f56-a289-f47340c5e942)
BEM circulation distribution

Another question, is it possible in fast or aerodyn to realize that the wind speed received by therotor varies along the spread?Meaning, not only the wind speed at the center of the hub is received, but a speed that varies along the Y-axis.

Best regards,

Roby

**Reply by ebranlard** [reply]
Hi, here are some elements of response:

1 & 4: BEM and OLAF come from different theories and different results are expected. 
2. Several things can cause the loads to be periodic: tilt angle, shear, tower influence, yaw. My guess is that you have a non zero tilt angle (ElastoDyn) and tower potential (AeroDyn).
3. Wind turbine signals are generally periodic (due to the effects mentioned above). Therefore it is always best to average over several period of revolutions. The functions averageDF and spanwisePostPro of pyFAST have this built in (see [here](https://github.com/OpenFAST/python-toolbox/blob/main/pyFAST/postpro/examples/Example_RadialPostPro.py))  

I recommend using [pyDatView ](https://github.com/ebranlard/pyDatView/releases)(or the script above) to visualize the average radial distribution of the quantities along the span. The wind speed is another output, just like "Gam" you can use "VDisx" "VDisy" to write the wind speed at different nodes to the output file. 

I hope that helps
